release-testing/secretless_test.go

```diff
@@ -0,0 +1,77 @@
+// +build release_test
+
+package main
+
+import (
+	"errors"
+	"os/exec"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"test/kv"
+)
+
+var storeClient *kv.StoreClient
+func init() {
+	var err error
+	storeClient, err = kv.DefaultStoreClient()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func callCommand(name string, args ...string) (string, error) {
+	out, err := exec.Command(
+		name,
+		args...
+	).Output()
+
+	outString := string(out)
+	if err != nil {
+		if exitErrr, ok := err.(*exec.ExitError); ok {
+			return outString, errors.New(string(exitErrr.Stderr))
+		}
+
+		return outString, err
+	}
+
+	return outString, nil
+}
+
+func TestSecretless(t *testing.T) {
+	t.Run("Consume conjur variable", func(t *testing.T) {
+		expectedValue := "abc123"
+
+		_, err := callCommand(
+			"bash",
+			"-ec",
+			// language=bash
+			`
+. ./executors.sh
+
+exec_conjur_client conjur variable values add test-app-secrets/username ` + expectedValue,
+		)
+		if !assert.NoError(t, err) {
+			return
+		}
+
+		out, err := callCommand(
+			"bash",
+			"-ec",
+			// language=bash
+			`
+. ./executors.sh
+
+exec_app bash -c '
+export http_proxy=http://localhost:8080
+
+curl -s "http://httpbin.org/anything" | jq -j ".headers.Authorization"
```

Comment:
I don't know if this would be useful or not, but it took me a little bit to figure out what's actually happening here (it being my first time digging into this secretless test).... But maybe it might help other readers to have a comment giving a quick overview of what's happening? Or maybe this is obvious. If I understand it correctly, the `http_proxy` is redirecting to secretless, secretless inserts the username as authen, and I guess the app is returning something/anything, so that the `curl -s "http://httpbin.org/anything" | jq -j ".headers.Authorization"` returns the username from the authorization in the response.

Reply:
I'll follow up with another PR to address these comments.